### PR TITLE
Update avatar models to use MToonMaterial

### PR DESCRIPTION
**Pull request by yaserOSource**
This PR enables use of MToonMaterial. 

Please look at the first 2 images for comparison. 

The last image is an art reference.

image
image
image

QA Checklist:

Look at the avatar face in the mirror, and up close with the camera, you should see some 'toon' shading.
Look at the avatar hair and body, you should see movement in the hair and other parts that might contain a 'springy' animation.

### CODE REVIEW

**Comment by avaer** [comment]
Can the art reference card be an app or something? Looks really useful.

**Comment by yaserOSource** [comment]
https://github.com/yaserOSource/avatar-test right here if you want to test them or upload them to app/public/avatars

**Comment by yaserOSource** [comment]
We can make it an app...too

**Comment by avaer** [comment]
What does https://github.com/yaserOSource/avatar-test have to do with the "art refrence" card?

**Comment by yaserOSource** [comment]
The reference spheres are amongst those models.

vian.glb
vian.vrm

**Comment by yaserOSource** [comment]
https://github.com/yaserOSource/material-debug-assets 

its these 2 assets, they were amongst the other avatars in that previously posted repo.

**Comment by avaer** [comment]
Got it, thanks for clarifying the naming.

**Comment by Vianvolaeus** [comment]
They're standalone models, so they can be tested easily across diff. engines for comparisons. The MToon one is equippable as a VRM avatar.

**Comment by skaljac** [comment]
It works for me
Screenshot (8)
Screenshot (9)
.

**Comment by amnaarshed4224** [comment]
This PR works fine with avatar's faces.

File: type_templates/vrm.js [hunk 1]
```diff
@@ -39,76 +36,15 @@ const _findMaterialsObjects = (o, name) => {
...
-          }
-          opts.gradientMap = gradientMap;
-          newMaterial = new THREE.MeshToonMaterial(opts);
-          materialMap.set(oldMaterial, newMaterial);
-        }
-        object.material = newMaterial;
-      }
-    }
+  const vrmImporter = new VRMMaterialImporter();
+  const materials = await vrmImporter.convertGLTFMaterials(o);
```
Comment (avaer) [inline type_templates/vrm.js:42]: As far as I can tell `convertGLTFMaterials` will already attach the converted materials.

https://github.com/pixiv/three-vrm/blob/96debe1ca3b3db8a462db0386af048f0d8339e87/packages/three-vrm/src/material/VRMMaterialImporter.ts#L99

So is the below loop even needed?

**Review by avaer — CHANGES_REQUESTED** [review]
One question, the rest looks like a great solution!

**Comment by yaserOSource** [comment]
This has been updated and the extra loop removed.

So its ready for merge.

**Comment by avaer** [comment]
This was not sufficiently tested and caused a crash when switching scenes.

In addition, I found race condition problems in this PR, where we do not wait for the shader to switch even though it is asynchronous.

**Comment by avaer** [comment]
image

**Comment by avaer** [comment]
This PR was reverted in https://github.com/webaverse/totum/commit/527994408e9d669e07943f32d01d495f79fd6a35

**Pull request merged**